File: apps/mobile/android/src/main/kotlin/xyz/scani/mobile/android/screens/SyncStatus.kt
package xyz.scani.mobile.android.screens

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

fun syncStatusLabel(lastSyncedAtMillis: Long?): String =
    if (lastSyncedAtMillis == null) "Never synced"
    else "Last synced: ${SimpleDateFormat("MMM d, HH:mm", Locale.getDefault()).format(Date(lastSyncedAtMillis))}"

File: apps/mobile/ios/iosApp/SyncStatus.swift
import Foundation

func syncStatusLabel(_ millis: Int64?) -> String {
    guard let millis else { return "Never synced" }
    let date = Date(timeIntervalSince1970: Double(millis) / 1000.0)
    let f = DateFormatter()
    f.dateFormat = "MMM d, HH:mm"
    return "Last synced: \(f.string(from: date))"
}
